src/components/PopupWithForm.js

```diff
@@ -20,10 +25,31 @@ class PopupWithForm extends Popup {
     });
     return inputValues;
   }
+  //this where the code for changing the save button to say "Saving..." goes
+
+  saveInfoListener() {
+    this._submitButton = this.popupForm.querySelector(".modal__button");
+    this._submitButton.textContent = "Saving...";
+  }
+
+  revertSaveButtonListener() {
+    this._submitButton = this.popupForm.querySelector(".modal__button");
+    this._submitButton.textContent = "Save";
+  }
 
   setEventListeners() {
     super.setEventListeners();
     this.popupForm.addEventListener("submit", () => {
+      this.saveInfoListener();
+      setTimeout(this._handleFormSubmit(this._getInputValues()), 5000);
+    });
+  }
+
+  updateProfileImageEventListeners() {
```

Comment:
Please, delete `updateProfileImageEventListeners`. It's a hardcode for an avatar modal. But `PopupWithForm` is **universal**


There should **not** be any code in universal functions (classes) that doesn’t suit every popup.